docs/source/traits_user_manual/observing.rst

```diff
@@ -0,0 +1,538 @@
+.. _observe-notification:
+
+==================================
+Trait Notification with |@observe|
+==================================
+
+Requesting trait attribute change notifications can be done in these
+ways:
+
+* Via class defintion: By decorating methods on the class with the |@observe|
+  decorator to indicate that they handle notification for specified attributes.
+* Via instance method: By calling |HasTraits.observe| instance method to establish (or remove) change
+  notification handlers.
+
+Via class definition
+--------------------
+
+Observers can be defined for every instance of a |HasTraits| subclass by
+applying the |@observe| decorator on an instance method. It has the following
+signature:
+
+.. py:decorator:: observe(expression[, post_init=False, dispatch="same"])
```

Comment:
Can this be made to use autodoc so that it doesn't get out of date if we change the API?

Reply:
autodoc ends up bringing the entire docstring here, which is too verbose, an alternative is to write our custom hooks for `autodoc` but that is probably an overkill. I agree with your concern, so I removed these now and hopefully the links to the API documentation is enough. I reworked these paragraphs to provide the same level of context for the subsequent sections. In particular, the mention of "expression" and "handler" are needed for later sections.